l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -669,100 +681,103 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
 
         address ctm = ctmAssetIdToAddress[_assetId];
         require(ctm != address(0), "BH: assetInfo 2");
-        require(settlementLayer[bridgeData.chainId] != block.chainid, "BH: already current SL");
+        require(settlementLayer[bridgehubData.chainId] != block.chainid, "BH: already current SL");
 
-        settlementLayer[bridgeData.chainId] = block.chainid;
-        chainTypeManager[bridgeData.chainId] = ctm;
-        baseTokenAssetId[bridgeData.chainId] = bridgeData.baseTokenAssetId;
+        settlementLayer[bridgehubData.chainId] = block.chainid;
+        chainTypeManager[bridgehubData.chainId] = ctm;
+        baseTokenAssetId[bridgehubData.chainId] = bridgehubData.baseTokenAssetId;
         // To keep `assetIdIsRegistered` consistent, we'll also automatically register the base token.
         // It is assumed that if the bridging happened, the token was approved on L1 already.
-        assetIdIsRegistered[bridgeData.baseTokenAssetId] = true;
+        assetIdIsRegistered[bridgehubData.baseTokenAssetId] = true;
 
-        address zkChain = getZKChain(bridgeData.chainId);
+        address zkChain = getZKChain(bridgehubData.chainId);
         bool contractAlreadyDeployed = zkChain != address(0);
         if (!contractAlreadyDeployed) {
-            zkChain = IChainTypeManager(ctm).forwardedBridgeMint(bridgeData.chainId, bridgeData.ctmData);
+            zkChain = IChainTypeManager(ctm).forwardedBridgeMint(bridgehubData.chainId, bridgehubData.ctmData);
             require(zkChain != address(0), "BH: chain not registered");
-            _registerNewZKChain(bridgeData.chainId, zkChain);
-            messageRoot.addNewChain(bridgeData.chainId);
+            _registerNewZKChain(bridgehubData.chainId, zkChain);
+            messageRoot.addNewChain(bridgehubData.chainId);
         }
 
-        IZKChain(zkChain).forwardedBridgeMint(bridgeData.chainData, contractAlreadyDeployed);
+        IZKChain(zkChain).forwardedBridgeMint(bridgehubData.chainData, contractAlreadyDeployed);
 
-        emit MigrationFinalized(bridgeData.chainId, _assetId, zkChain);
+        emit MigrationFinalized(bridgehubData.chainId, _assetId, zkChain);
     }
 
     /// @dev IL1AssetHandler interface, used to undo a failed migration of a chain.
-    /// @param _chainId the chainId of the chain
+    // / @param _chainId the chainId of the chain
     /// @param _assetId the assetId of the chain's CTM
     /// @param _data the data for the recovery.
     function bridgeRecoverFailedTransfer(
-        uint256 _chainId,
+        uint256,
         bytes32 _assetId,
         address _depositSender,
         bytes calldata _data
     ) external payable override onlyAssetRouter onlyL1 {
-        BridgehubBurnCTMAssetData memory ctmAssetData = abi.decode(_data, (BridgehubBurnCTMAssetData));
+        BridgehubBurnCTMAssetData memory bridgehubData = abi.decode(_data, (BridgehubBurnCTMAssetData));
 
-        delete settlementLayer[_chainId];
```

Comment:
I want to mark that this was wrong. ( in case I'm wrong)

Reply:
yes, you are totally right, thank you